onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -253,7 +253,7 @@ class PlannerImpl {
     // TODO: This should probably be defined to be the equality operator on TensorShapeProto.
     namespace on = ONNX_NAMESPACE;
     int rank1 = shape1.dim_size();
-    if (shape2.dim_size() != rank1) return false;
+    if (rank1 == 0 || shape2.dim_size() != rank1) return false;
```

Comment:
This function itself is only for comparing two shapes. if both shape are empty, should they still be treated as "Same" ?

Reply:
Also, shape.dim_size is returning the rank of the shape. a shape proto with value {2,2,0} will return dim_size of 3.  IIUC, you are looking to check if a shape tensor has any 0 as the dim_value?

Reply:
Here we are trying to address the issue of reused buffer. In this case, the dim_size() is 0 in allocation planner, and it just 
select one from freelist_ that matches the shape (also dime_size() == 0). But at execution frame, the shape is {14, 428}, thus 
exception thrown. Therefore we don't want it to be "AllocKind::kReuse", instead to be "AllocKind::kAllocate".
Any other suggestion/idea how to workaround this issue?  
 	
RuntimeError: Error in execution: Non-zero status code returned while running SoftmaxCrossEntropyLoss node. 
Name:'SoftmaxCrossEntropyLoss_1819' Status Message: /bert_ort/ettao/onnx1/onnxruntime/onnxruntime/core/framework/op_kernel.cc:52 
OrtValue* onnxruntime::OpKernelContext::OutputMLValue(int, const onnxruntime::TensorShape&, size_t) status.IsOK() was false. 
Shape mismatch attempting to re-use buffer. {} != {14,428}. Validate usage of dim_value (values should be > 0) and 
dim_param (all values with the same string should equate to the same size) in shapes in the model.